src/app/loadout/SavedModCategory.tsx

```diff
@@ -0,0 +1,74 @@
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
+import { PluggableInventoryItemDefinition } from 'app/inventory/item-types';
+import LockedModIcon from 'app/loadout-builder/filter/LockedModIcon';
+import React, { useCallback, useState } from 'react';
+import { AddButton } from './Buttons';
+import styles from './SavedMods.m.scss';
+
+interface Props {
+  defs: D2ManifestDefinitions;
+  /** A list of mods that all have the same plugCategoryHash. */
+  mods: PluggableInventoryItemDefinition[];
+  /** Removes a mod from the loadout via the mods item hash. */
+  onRemove(itemHash: number): void;
+  /** Opens the mod picker sheet with a supplied query to filter the mods. */
+  onOpenModPicker(query?: string): void;
+}
+
+/**
+ * A component for displaying a group of mods categorised by their plugCategoryHash.
+ *
+ * It allows the mods to be added to and removed from the loadout.
+ */
+function SavedModCategory({ defs, mods, onRemove, onOpenModPicker }: Props) {
+  const [width, setWidth] = useState<number | undefined>();
+  const firstMod = mods.length && mods[0];
+
+  const widthSetter = useCallback(
```

Comment:
What's this for?

Reply:
For this ellipsis effect. I tried a few ways to do it without setting the width on the heading itself but couldn't figure something out. It's entirely possible I just wasn't hooking it up correctly with only css.

![image](https://user-images.githubusercontent.com/7344652/113782083-f8235e00-9774-11eb-8fc9-fefd267b10cb.png)

Reply:
I'm sure there's a pure CSS way to do it but it's probably harder than what you did.

Reply:
ok I will do a bit of reading and see if I can figure it out

Reply:
Sorted it out using css, just involved another div and some flex.